edxadapt/static/js/src/edxadapt.js

```diff
@@ -43,13 +44,17 @@ function EdxAdaptXBlock(runtime, element) {
             })
             .done(function() {
                 console.log("User Configured");
-                setStatusOk();
             })
             .fail(function() {
                 console.log('Failed to configure skill "'+skill+'" for user "'+studentId+'"');
+                ready_to_enroll = false;
                 setStatusNok();
             });
         });
+        if (ready_to_enroll) {
```

Comment:
This way it will always trigger `registerUser()` because `ready_to_enroll = false` is set asynchronously after this line was executed. Due to network latency.